subprojects/jacoco/src/main/java/org/gradle/internal/jacoco/AntJacocoCheck.java

```diff
@@ -44,10 +49,31 @@ public AntJacocoCheck(IsolatedAntBuilder ant) {
         super(ant);
     }
 
+    public JacocoCheckResult execute(FileCollection classpath, final String projectName,
+                                     final FileCollection allClassesDirs, final FileCollection allSourcesDirs,
+                                     final FileCollection executionData, final JacocoViolationRulesContainer violationRules) {
+        final JacocoCheckResult jacocoCheckResult = new JacocoCheckResult();
+
+        configureAntReportTask(classpath, new Action<GroovyObjectSupport>() {
+            @Override
+            public void execute(GroovyObjectSupport antBuilder) {
+                try {
+                    invokeJacocoReport(antBuilder, projectName, allClassesDirs, allSourcesDirs, executionData, violationRules);
+                } catch (Exception e) {
+                    String violations = getViolations(antBuilder);
+                    jacocoCheckResult.setSuccess(false);
+                    jacocoCheckResult.setFailureMessage(violations != null ? violations : e.getMessage());
+                }
+            }
+        });
+
+        return jacocoCheckResult;
+    }
+
     @Override
     protected void configureReport(final GroovyObjectSupport antBuilder, final JacocoViolationRulesContainer violationRules) {
         if (!violationRules.getRules().isEmpty()) {
-            Map<String, Object> checkArgs = ImmutableMap.<String, Object>of("failonviolation", !violationRules.isFailOnViolation());
+            Map<String, Object> checkArgs = ImmutableMap.<String, Object>of("failonviolation", violationRules.isFailOnViolation(), "violationsproperty", VIOLATIONS_ANT_PROPERTY);
```

Comment:
An exception would still be thrown from Ant if you provide invalid validation rules _and_ make `failonviolation = false`. For more info see test case `JacocoPluginCoverageVerificationIntegrationTest.Ant task reports error for unknown field value`.